pypreprocess/time_diff.py

```diff
@@ -0,0 +1,261 @@
+
+    # loop over scans: increment statistics
+    for vol_index in range(0, T - 1):
+        current_vol = arr[..., vol_index + 1]  # shape vol_shape
+        vol_mean[vol_index + 1] = current_vol.mean()
+        squared_diff = (current_vol - last_vol) ** 2
+        diff_mean_vol += squared_diff
+        slice_squared_differences[vol_index] = squared_diff.mean(0).mean(0)
+        # check whether we have found a highest-diff slice
+        larger_diff = slice_squared_differences[vol_index] > slice_diff_max
+        if any(larger_diff):
+            slice_diff_max[larger_diff] =\
+                slice_squared_differences[vol_index][larger_diff]
+            slice_diff_max_vol[..., larger_diff] =\
+                squared_diff[..., larger_diff]
+        last_vol = current_vol
+    vol_squared_differences = slice_squared_differences.mean(1)
+    diff_mean_vol /= (T - 1)
+
+    # Return the outputs as images
+    affine = img.get_affine()
+    diff2_mean_vol = nib.Nifti1Image(diff_mean_vol, affine)
+    slice_diff2_max_vol = nib.Nifti1Image(slice_diff_max_vol, affine)
+    return {'volume_mean_diff2': vol_squared_differences,
+            'slice_mean_diff2': slice_squared_differences,
+            'volume_means': vol_mean,
+            'diff2_mean_vol': diff2_mean_vol,
+            'slice_diff2_max_vol': slice_diff2_max_vol}
+
+
+def plot_tsdiffs(results, use_same_figure=True):
+    ''' Plotting routine for time series difference metrics
+
+    Requires matplotlib
+
+    Parameters
+    ----------
+    results : dict
+        Results of format returned from
+        :func:`pypreprocess.time_diff.time_slice_diff`
+
+    use_same_figure : bool
+        Whether to put all the plots on the same figure. If False, one
+        figure will be created for each plot.
+
+    '''
+    import matplotlib.pyplot as plt
+
+    T = len(results['volume_means'])
+    S = results['slice_mean_diff2'].shape[1]
+    mean_means = np.mean(results['volume_means'])
+    scaled_slice_diff = results['slice_mean_diff2'] / mean_means
+    n_plots = 6
+
+    if use_same_figure:
+        fig, axes = plt.subplots((n_plots + 1) // 2, 2)
+        # Slightly easier to flatten axes to treat the
+        # use_same_figure=False case in a similar fashion
+        axes = axes.T.reshape(-1)
+        fig.set_size_inches(12, 6, forward=True)
+        fig.subplots_adjust(top=0.97, bottom=0.08, left=0.1, right=0.98,
+                            hspace=0.3, wspace=0.18)
+    else:
+        axes = [plt.figure().add_subplot(111)
+                for _ in range(n_plots - 2)]
+
+    def xmax_labels(ax, val, xlabel, ylabel):
+        xlims = ax.axis()
+        ax.axis((0, val) + xlims[2:])
+        ax.set_xlabel(xlabel)
+        ax.set_ylabel(ylabel)
+
+    iter_axes = iter(axes)
+
+    # plot of mean volume variance
+    ax = next(iter_axes)
+    ax.plot(results['volume_mean_diff2'] / mean_means)
+    xmax_labels(ax, T - 1, 'Difference image number', 'Scaled variance')
```

Comment:
@bthirion all the xmax_labels calls are where you can improve of the axes titles.